qualang_tools/config/helper_tools.py

```diff
@@ -0,0 +1,390 @@
+        self, element: str, operation_name: str
+    ) -> List or List[List]:
+        """
+        Get the waveforms corresponding to the given element and operation.
+
+        :param element: name of the element to get the waveforms from. Must be defined in the config.
+        :param operation_name: name of the operation to get the waveforms from. Must be defined in the config.
+        :return: list of the corresponding waveforms.
+        """
+
+        # Check inputs
+        if element not in self.data["elements"].keys():
+            raise KeyError(f"The element '{element}' in not defined in the config.")
+        if operation_name not in self.data["elements"][element]["operations"].keys():
+            raise KeyError(
+                f"The operation '{operation_name}' in not defined in the config for the element '{element}'."
+            )
+
+        # Get the waveforms
+        pulse = self.get_pulse_from_op(element, operation_name)
+        if "mixInputs" in self.data["elements"][element]:
+            waveform_i = self.data["waveforms"][pulse["waveforms"]["I"]]
+            if waveform_i["type"] == "arbitrary":
+                waveform_i = waveform_i["samples"]
+            else:
+                waveform_i = [waveform_i["sample"]] * pulse["length"]
+
+            waveform_q = self.data["waveforms"][pulse["waveforms"]["Q"]]
+            if waveform_q["type"] == "arbitrary":
+                waveform_q = waveform_q["samples"]
+            else:
+                waveform_q = [waveform_q["sample"]] * pulse["length"]
+            return waveform_i, waveform_q
+        else:
+            waveform = self.data["waveforms"][pulse["waveforms"]["single"]]
+            if waveform["type"] == "arbitrary":
+                waveform = waveform["samples"]
+            else:
+                waveform = [waveform["sample"]] * pulse["length"]
+            return waveform
+
+    def get_pulse_from_op(self, element: str, operation_name: str) -> dict:
+        """
+        Get the pulse corresponding to the given element and operation.
+
+        :param element: name of the element to get the waveforms from. Must be defined in the config.
+        :param operation_name: name of the operation to get the waveforms from. Must be defined in the config.
+        :return: the corresponding pulse as a dictionary.
+        """
+
+        # Check inputs
+        if element not in self.data["elements"].keys():
+            raise KeyError(f"The element '{element}' in not defined in the config.")
+        if operation_name not in self.data["elements"][element]["operations"].keys():
+            raise KeyError(
+                f"The operation '{operation_name}' in not defined in the config for the element '{element}'."
+            )
+
+        return self.data["pulses"][
+            self.data["elements"][element]["operations"][operation_name]
+        ]
+
+    def update_op_amp(self, element: str, operation_name: str, amp: float):
```

Comment:
This changes the waveform amplitude, which can affect other elements as well!
This has to be at least noted... Options:
1. Just note it.
2. Look to see if anyone else is using that waveform (search through the pulses/operations) and if so, give a warning/error/option 3 from below
3. Copy the waveform and create a new one with the new amplitude

We can also have a flag indicating the behavior (between 1 & 3, or between 2 & 3 if we decide to do it)

Reply:
This is a good point, thanks!
Now I check if it used elsewhere and I added a flag to force the update (default is False)